tests/components/test_button.py

```diff
@@ -0,0 +1,81 @@
+"""Tests for the Button Classes."""
+from time import sleep, time
+from typing import List, Type
+
+from j5.backends import Backend
+from j5.boards import Board
+from j5.components.button import Button, ButtonInterface
+
+
+class MockButtonDriver(ButtonInterface):
+    """A testing driver for the button component."""
+
+    def get_button_state(self, board: Board, identifier: int) -> bool:
+        """Get the state of the button."""
+        return False  # Never pushed.
+
+    def wait_until_button_pressed(self, board: Board, identifier: int) -> bool:
+        """Wait until the button was pressed."""
+        sleep(0.2)  # The mock driver always presses the button after 0.2s
+
+
+class MockButtonBoard(Board):
+    """A testing board for the button."""
+
+    @property
+    def name(self) -> str:
+        """The name of this board."""
+        return "Testing Button Board"
+
+    @property
+    def serial(self) -> str:
+        """The serial number of this board."""
+        return "SERIAL"
+
+    def make_safe(self):
+        """Make this board safe."""
+        pass
+
+    @staticmethod
+    def supported_components() -> List[Type['Component']]:
+        """List the types of component that this Board supports."""
+        return [Button]
+
+    @staticmethod
+    def discover(backend: Backend) -> List['Board']:
+        """Detect all of the boards on a given backend."""
+        return []
+
+
+def test_button_interface_implementation():
+    """Test that we can implement the button interface."""
+    MockButtonDriver()
+
+
+def test_button_instantiation():
+    """Test that we can instantiate a button."""
+    Button(0, MockButtonBoard(), MockButtonDriver())
+
+
+def test_button_state():
+    """Test that we can get the state of the button."""
+    button = Button(0, MockButtonBoard(), MockButtonDriver())
+
+    assert not button.is_pressed
```

Comment:
Since this function has direct ownership of the `MockButtonDriver`, I think it would make sense to have `MockButtonDriver.get_button_state` return an attribute of the `MockButtonDriver` instance which is set to different values by this function. This would allow the test function to do a better job of verifying that `MockButtonDriver.get_button_state` propagates to `Button.is_pressed` by testing more than one value.

Reply:
Additionally, part of me wants to change this assertion to `assert button.is_pressed is False` to verify that we're returning a boolean and not anything else that happens to be false-y (e.g. an empty string). I don't know whether this test scenario is useful to consider though. What are your thoughts?

Reply:
> Additionally, part of me wants to change this assertion to `assert button.is_pressed is False` to verify that we're returning a boolean and not anything else that happens to be false-y (e.g. an empty string). I don't know whether this test scenario is useful to consider though. What are your thoughts?

I feel like this is what a type checker does, but have done it for you anyway.